File: front/src/components/authentication/dtos/business-auth.ts

import { RootBase } from "shared/entities-dtos/root-base";

import { UserAccountAuthDto } from "../../authentication/dtos/user-account-auth-dto";
import { CompanyAuth } from "components/company/dtos/company-auth";

export class BusinessAuth extends RootBase {
  name!: string;
  businessProfileId!:string;
  usersAccounts!: UserAccountAuthDto[];
  companies!: CompanyAuth[];

}

File: back/Domain/Entities/System/Businesses/BusinessAuth.cs
using Domain.Entities.Authentication;
using Domain.Entities.Shared;
using Domain.Entities.System.Businesses.extends;
using Domain.Entities.System.Companies;

namespace Domain.Entities.System.Businesses;

public class BusinessAuth : BusinessBaseDb
{
    public required string Name { get; set; }
    public required string BusinessProfileId { get; set; }
    public ICollection<UserAccount> UsersAccounts { get; set; } = new List<UserAccount>();
    public ICollection<CompanyAuth> Companies { get; set; } = new List<CompanyAuth>();
    
}

